File: base/src/main/java/com/mindera/skeletoid/generic/DebugTools.java
package com.mindera.skeletoid.generic;

import android.support.annotation.VisibleForTesting;

import com.mindera.skeletoid.logs.LOG;

import java.util.Map;

/**
 * Class to provide debug only methods and utilities that should NOT be used in production
 */
public class DebugTools {

    @VisibleForTesting
    DebugTools() {
        throw new UnsupportedOperationException();
    }

    public static void printAllStackTraces(Class<?> clazz) {

        LOG.d(clazz.toString(), "DUMPING ALL STACK TRACES");

        Map<Thread, StackTraceElement[]> liveThreads = Thread.getAllStackTraces();
        for (Map.Entry<Thread, StackTraceElement[]> entry : liveThreads.entrySet()) {
            LOG.d(clazz.toString(), "Thread " + entry.getKey().getName());
            StackTraceElement[] traceElements = entry.getValue();
            for (StackTraceElement traceElement : traceElements) {
                LOG.d(clazz.toString(), "at " + traceElement);
            }
        }
    }
}

File: base/src/main/java/com/mindera/skeletoid/generic/DebugTools.kt
package com.mindera.skeletoid.generic

import com.mindera.skeletoid.logs.LOG

/**
 * Class to provide debug only methods and utilities that should NOT be used in production
 */
object DebugTools {
    @JvmStatic
    fun printAllStackTraces(clazz: Class<*>) {
        LOG.d(clazz.toString(), "DUMPING ALL STACK TRACES")
        val liveThreads = Thread.getAllStackTraces()
        for ((key, traceElements) in liveThreads) {
            LOG.d(clazz.toString(), "Thread " + key.name)
            for (traceElement in traceElements) {
                LOG.d(clazz.toString(), "at $traceElement")
            }
        }
    }
}